### PR TITLE
Global CONSTs

### PR DESCRIPTION
**Pull request by DevRozaDev**
+ alephium web3, stack bump
+ alephium stack update script
+ make error codes global, move to a single location (following this discussion: https://github.com/invariant-labs/protocol-alephium/pull/115#discussion_r1692791345)
+ move types to a single location alongside consts relevant to them (same discussion as above)
+ use denominator constant instead of `one` where applicable
+ replace uses of `u256Max!()` in Decimal contract with `MAX_U256`
+ display contracts' bytecode length
+ replace log getters with consts

### CODE REVIEW

File: contracts/invariant.ral [hunk 1]
```diff
@@ -1,25 +1,5 @@
...
-    targetSqrtPrice: SqrtPrice,
-    fee: TokenAmount
-}
-
-struct QuoteResult {
-    amountIn: TokenAmount,
-    amountOut: TokenAmount,
-    targetSqrtPrice: SqrtPrice
-}
+const MaxFeeTiers = 32
```
Comment (Sniezka1927) [inline contracts/invariant.ral:2]:
```suggestion
const MAX_FEE_TIERS = 32
```

File: src/math.ts [hunk 2]
```diff
@@ -290,30 +290,30 @@ export const calculateSqrtPriceAfterSlippage = (
...
+  const priceWithSlippage = price * multiplier * PERCENTAGE_DENOMINATOR
+  const sqrtPriceWithSlippage = priceToSqrtPrice(priceWithSlippage) / PERCENTAGE_DENOMINATOR
 
   return sqrtPriceWithSlippage
 }
 
 export const toLiquidity = (value: bigint, offset = 0n) => {
-  if (offset >= LiquidityScale) throw new Error(`offset must be less than ${LiquidityScale}`)
-  return value * 10n ** (LiquidityScale - offset)
+  if (offset >= LIQUIDITY_SCALE) throw new Error(`offset must be less than ${LIQUIDITY_SCALE}`)
```
Comment (Sniezka1927) [inline src/math.ts:302]: We should accept offset equal to scale

File: package.json [hunk 3]
```diff
@@ -28,7 +28,7 @@
     "clean:windows": "node -e \"if (process.platform === 'win32') process.exit(1)\" || , if exist dist rmdir /Q /S dist",
     "devnet:start": "npx --yes @alephium/cli@latest devnet start",
     "devnet:stop": "npx --yes @alephium/cli@latest devnet stop",
-    "compile": "npx cli compile",
+    "compile": "npx cli compile && npx ts-node stats",
```
Comment (Sniezka1927) [inline package.json:31]: imo it will be better to run "npx cli compile && npm run stats", so we maintain single source of running the script

**Review by Sniezka1927 — CHANGES_REQUESTED** [review]

File: contracts/invariant.ral [hunk 4]
```diff
@@ -1,26 +1,6 @@
...
-    fee: TokenAmount
-}
-
-struct QuoteResult {
-    amountIn: TokenAmount,
-    amountOut: TokenAmount,
-    targetSqrtPrice: SqrtPrice
-}
-
+const MAX_FEE_TIERS = 32
```
Comment (Sniezka1927) [inline contracts/invariant.ral:3]: Shouldn't the const be in `constants.ral` as well as other structs moved to `structures/` directory?

**Review by Sniezka1927 — APPROVED** [review]

**Review by wojciech-cichocki — APPROVED** [review]

**Pull request merged**